Application/src/main/java/com/shareyourproxy/api/domain/model/ChannelType.java

```diff
@@ -22,41 +23,40 @@
     URL(4, "URL", R.raw.ic_link, WHITE),
     Facebook(5, "Facebook", R.raw.ic_facebook, WHITE),
     Twitter(6, "Twitter", R.raw.ic_twitter, WHITE),
-    Meerkat(7, "Meerkat", R.raw.ic_meerkat, NO_COLOR),
-    Snapchat(8, "Snapchat", R.raw.ic_snapchat, NO_COLOR),
-    Spotify(9, "Spotify", R.raw.ic_spotify, NO_COLOR),
-    Reddit(10, "Reddit", R.raw.ic_reddit, NO_COLOR),
-    Linkedin(11, "Linkedin", R.raw.ic_linkedin, NO_COLOR),
-    FBMessenger(12, "Facebook Messenger", R.raw.ic_facebook_messenger, NO_COLOR),
-    Hangouts(13, "Hangouts", R.raw.ic_google_hangouts, NO_COLOR),
-    Whatsapp(14, "Whats App", R.raw.ic_whatsapp, NO_COLOR),
-    Yo(15, "Yo", R.raw.ic_yo, NO_COLOR),
-    Googleplus(16, "Google Plus", R.raw.ic_google_plus, NO_COLOR),
-    Github(17, "Github", R.raw.ic_github, NO_COLOR),
-    Address(18, "Address", R.raw.ic_address, NO_COLOR),
-    Slack(19, "Slack", R.raw.ic_slack, NO_COLOR),
-    Youtube(20, "Youtube", R.raw.ic_youtube, NO_COLOR),
-    Instagram(21, "Instagram", R.raw.ic_instagram, NO_COLOR),
-    Tumblr(22, "Tumblr", R.raw.ic_tumblr, NO_COLOR),
-    Ello(23, "Ello", R.raw.ic_ello, NO_COLOR),
-    Venmo(24, "Venmo", R.raw.ic_venmo, NO_COLOR),
-    Periscope(25, "Periscope", R.raw.ic_periscope, NO_COLOR),
-    Medium(26, "Medium", R.raw.ic_medium, NO_COLOR),
-    Soundcloud(27, "Soundcloud", R.raw.ic_soundcloud, NO_COLOR),
-    Skype(28, "Skype", R.raw.ic_skype, NO_COLOR);
-
+    Meerkat(7, "Meerkat", R.raw.ic_meerkat, null),
+    Snapchat(8, "Snapchat", R.raw.ic_snapchat, null),
+    Spotify(9, "Spotify", R.raw.ic_spotify, null),
+    Reddit(10, "Reddit", R.raw.ic_reddit, null),
+    Linkedin(11, "Linkedin", R.raw.ic_linkedin, null),
+    FBMessenger(12, "Facebook Messenger", R.raw.ic_facebook_messenger, null),
+    Hangouts(13, "Hangouts", R.raw.ic_google_hangouts, null),
+    Whatsapp(14, "Whats App", R.raw.ic_whatsapp, null),
+    Yo(15, "Yo", R.raw.ic_yo, null),
+    Googleplus(16, "Google Plus", R.raw.ic_google_plus, null),
+    Github(17, "Github", R.raw.ic_github, null),
+    Address(18, "Address", R.raw.ic_address, null),
+    Slack(19, "Slack", R.raw.ic_slack, null),
+    Youtube(20, "Youtube", R.raw.ic_youtube, null),
+    Instagram(21, "Instagram", R.raw.ic_instagram, null),
+    Tumblr(22, "Tumblr", R.raw.ic_tumblr, null),
+    Ello(23, "Ello", R.raw.ic_ello, null),
+    Venmo(24, "Venmo", R.raw.ic_venmo, null),
+    Periscope(25, "Periscope", R.raw.ic_periscope, null),
+    Medium(26, "Medium", R.raw.ic_medium, null),
+    Soundcloud(27, "Soundcloud", R.raw.ic_soundcloud, null),
+    Skype(28, "Skype", R.raw.ic_skype, null);
 
     private final int weight;
     private final String label;
     private final int resId;
-    private final int resColor;
+    private final Integer resColor;
 
     /**
      * Constructor.
      *
      * @param label name of newChannel
      */
-    ChannelType(int weight, String label, int resId, int resColor) {
```

Comment:
I did this because some ChannelType icons need the concept of "NO_COLOR", or a null value. This is different logic than selecting Color.TRANSPARENT, which produces a weird svg when we're constructing the icons. Before I was using int with the value NO_COLOR = Integer.MIN_VALUE as my logic switch. I don't like either solution to be honest, but using null to switch rather than a random constant seems better for now.